pkg/providerplugin/providerplugin.go

```diff
@@ -0,0 +1,40 @@
+package providerplugin
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/SovereignCloudStack/csmctl/pkg/clusterstack"
+)
+
+func CheckProviderExecutable(config *clusterstack.CsmctlConfig) (path string, err error) {
+	pluginName := "csmctl-" + config.Config.Provider.Type
+	_, err = os.Stat(pluginName)
+	if err == nil {
+		path, err := filepath.Abs(pluginName)
+		if err != nil {
+			return "", err
+		}
+		return path, err
+	}
+	path, err = exec.LookPath(pluginName)
+	if err != nil {
+		return "", fmt.Errorf("could not find plugin %s in $PATH or current working directory", pluginName)
+	}
+	return path, nil
+}
+
+func CreateNodeImages(config *clusterstack.CsmctlConfig, clusterStackPath string, clusterStackReleaseDir string) error {
+	path, err := CheckProviderExecutable(config)
```

Comment:
I think you don't need to repeat this

Reply:
I think I will rename the method to GetProviderExecutable(). Then it is less confusing. I need to get the `path`. And coding this twice is doable, but error-prone.

Reply:
ah okay. I only saw the "check", so renaming it sounds like a good idea!

Reply:
done.